### PR TITLE
8341692: Implement JEP 490: ZGC: Remove the Non-Generational Mode

### PR DESCRIPTION
**Pull request by xmas92**
This is the implementation task for `JEP 490: ZGC: Remove the Non-Generational Mode`. See the JEP for details. [JDK-8335850](https://bugs.openjdk.org/browse/JDK-8335850)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a JEP request to be targeted
- [x] Change must be properly reviewed (3 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 2 [Authors](https://openjdk.org/bylaws#author))
- [x] Change requires CSR request [JDK-8341693](https://bugs.openjdk.org/browse/JDK-8341693) to be approved

### Issues
 * [JDK-8341692](https://bugs.openjdk.org/browse/JDK-8341692): Implement JEP 490: ZGC: Remove the Non-Generational Mode (**Sub-task** - P4)
 * [JDK-8335850](https://bugs.openjdk.org/browse/JDK-8335850): JEP 490: ZGC: Remove the Non-Generational Mode (**JEP**)
 * [JDK-8341693](https://bugs.openjdk.org/browse/JDK-8341693): Implement JEP 490: ZGC: Remove the Non-Generational Mode (**CSR**)

### Reviewers
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)
 * [Erik Österlund](https://openjdk.org/census#eosterlund) (@fisk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) Review applies to [e58b4c5a](https://git.openjdk.org/jdk/pull/21401/files/e58b4c5a800f7e66124c20b26f6b6a93ad8934f7)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) Review applies to [e58b4c5a](https://git.openjdk.org/jdk/pull/21401/files/e58b4c5a800f7e66124c20b26f6b6a93ad8934f7)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) Review applies to [e58b4c5a](https://git.openjdk.org/jdk/pull/21401/files/e58b4c5a800f7e66124c20b26f6b6a93ad8934f7)
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21401/head:pull/21401` \
`$ git checkout pull/21401`

Update a local copy of the PR: \
`$ git checkout pull/21401` \
`$ git pull https://git.openjdk.org/jdk.git pull/21401/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21401`

View PR using the GUI difftool: \
`$ git pr show -t 21401`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21401.diff">https://git.openjdk.org/jdk/pull/21401.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21401#issuecomment-2402104028)

### CODE REVIEW

**Review by magicus — APPROVED** [review]
Build changes look good.

Are you on a direct track towards winning over the title of Dr Deprecator? ;-)

/reviewers 3

**Comment by stefank** [comment]
Looks good to me!

File: src/hotspot/share/runtime/arguments.cpp [hunk 1]
```diff
@@ -521,7 +520,7 @@ static SpecialFlag const special_jvm_flags[] = {
   // -------------- Obsolete Flags - sorted by expired_in --------------
 
   { "MetaspaceReclaimPolicy",       JDK_Version::undefined(), JDK_Version::jdk(21), JDK_Version::undefined() },
-
+  { "ZGenerational",                JDK_Version::jdk(23), JDK_Version::jdk(24), JDK_Version::undefined() },
```
Comment (stefank) [inline src/hotspot/share/runtime/arguments.cpp:523]: FTR: This line depends on what version the JEP gets targeted to.

**Review by fisk — APPROVED** [review]
Looks good.

**Review by prrace — APPROVED** [review]
the 2D/AWT test changes are fine. I've not looked at anything else.

**Review by plummercj — APPROVED** [review]
The serviceability related changes look good. I suppose the 6 or so SA bugs filed against the non-generational ZGC can be closed out. I have a note to myself to go through them and close if appropriate. We have [JDK-8307393](https://bugs.openjdk.org/browse/JDK-8307393) filed for the lack of SA support for generational ZGC, and that is probably the only SA ZGC related issue we need to keep around.

**Review by dholmes-ora — APPROVED** [review]
I skimmed everything and it seems okay to me.

Thanks

**Review by stefank — APPROVED** [review]

**Review by stefank — APPROVED** [review]

**Review by fisk — APPROVED** [review]

**Review by magicus — APPROVED** [review]

**Comment by xmas92** [comment]
Thanks for the reviews. 
/integrate

**Pull request closed without merging**